server/src/main/java/com/flipkart/varadhi/authn/CustomAuthenticationHandler.java

```diff
@@ -0,0 +1,129 @@
+    private static final String DEFAULT_ORG = "SYSTEM";
+    private Authenticator authenticator;
+    private OrgResolver orgResolver;
+
+    /**
+     * Provides a custom authentication handler that uses the configured Authenticator implementation.
+     * This method initializes the authenticator from the provided configuration and returns this handler
+     * instance configured with that authenticator.
+     *
+     * @param vertx       The Vertx instance
+     * @param jsonObject  Configuration parameters containing authenticator provider class name and settings
+     * @param orgResolver Organization resolver (not used in custom authentication)
+     * @param meterRegistry for registering metrics
+     * @return AuthenticationHandler
+     * @throws InvalidConfigException if the authenticator provider class cannot be loaded or initialized
+     */
+
+
+    @Override
+    public AuthenticationHandler provideHandler(
+        Vertx vertx,
+        JsonObject jsonObject,
+        OrgResolver orgResolver,
+        MeterRegistry meterRegistry
+    ) {
+        return provideHandler(vertx, jsonObject.mapTo(AuthenticationConfig.class), orgResolver, meterRegistry);
+    }
+
+    private AuthenticationHandler provideHandler(
+        Vertx vertx,
+        AuthenticationConfig authenticationConfig,
+        OrgResolver orgResolver,
+        MeterRegistry meterRegistry
+    ) {
+        try {
+            if (authenticationConfig.getAuthenticatorClassName().isEmpty()) {
+                throw new InvalidConfigException("Empty/Null Authenticator class name");
+            }
+
+            Class<?> providerClass = Class.forName(authenticationConfig.getAuthenticatorClassName());
+            if (!Authenticator.class.isAssignableFrom(providerClass)) {
+                throw new InvalidConfigException(
+                    "Provider class " + providerClass.getName() + " does not implement Authenticator interface"
+                );
+            }
+            authenticator = (Authenticator)providerClass.getDeclaredConstructor().newInstance();
+
+            try {
+                authenticator.init(authenticationConfig, meterRegistry);
+            } catch (Exception e) {
+                throw new InvalidConfigException("Failed to initialize authenticator: " + e.getMessage(), e);
+            }
+
+        } catch (ClassNotFoundException e) {
+            throw new InvalidConfigException(
+                "Authentication provider class not found: " + authenticationConfig.getAuthenticatorClassName(),
+                e
+            );
+        } catch (ReflectiveOperationException e) {
+            throw new InvalidConfigException("Failed to create authentication provider", e);
+        }
+
+        return new CustomAuthenticationHandler(authenticator, orgResolver);
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+
+        Org org = orgResolver.resolve(DEFAULT_ORG);
```

Comment:
why are  we resolving the DEFAULT ORG?

Reply:
we are not using this org info. not sure if we should be reading it from anywhere.